backend/controllers/retentionReasons.js

```diff
@@ -83,7 +80,7 @@ const retentionReasonsFactory = (elite2Api, dataComplianceApi, logError) => {
         retentionReasons: reasons.filter(reason => reason.reasonCode),
       }
 
-      dataComplianceApi
+      return await dataComplianceApi
```

Comment:
Bit confusing mixing `await` and promises..

can get rid of the `then` and `catch` if using await. (any exception would be caught by the `} catch (error) {`)
```js
    await dataComplianceApi.putOffenderRetentionRecord(res.locals, offenderNo, optionsSelected, version)
    return res.redirect(getOffenderUrl(offenderNo))
       
  } catch (error) {
      return renderError(req, res, error)
    }
```